src/peer.rs

```diff
@@ -190,50 +199,68 @@ impl RemotePeer {
             MessageType::Choke => {
                 // peer has choked us
                 self.peer_choking = true;
+                info!("Peer {:#?} has choked us", self.peer);
             }
             MessageType::UnChoke => {
                 // peer has unchoked us
                 self.unchoke_us().await;
+                info!("Peer {:#?} has unchoked us", self.peer);
             }
             MessageType::Interested => {
                 // peer is interested in us
                 self.peer_interested = true;
+                info!("Peer {:#?} is interested in us", self.peer);
             }
             MessageType::NotInterested => {
                 // peer is not interested in us
                 self.peer_interested = false;
+                info!("Peer {:#?} is uninterested in us", self.peer);
             }
             MessageType::Have => {
                 // peer has piece <piece_index>
                 // sent after piece is downloaded and verified
                 let piece_index = slice_to_u32_msb(&bt_msg.payload[0..4]);
                 self.piece_lookup.mark_piece(piece_index);
+
+                info!("Peer {:#?} has confirmed that they have piece with piece-index {}", self.peer, piece_index);
             }
             MessageType::Bitfield => {
                 // info about which pieces peer has
                 // only sent right after handshake, and before any other msg (so optional)
                 self.piece_lookup = BitField(bt_msg.payload.clone());
+                info!("Peer {:#?} has given us its bitfield", self.peer);
```

Comment:
would be nice if you can implement Debug/Display for bitfield indicating what they have in a clean format